src/js/components/Form/Form.js

```diff
@@ -300,9 +301,7 @@ const Form = forwardRef(
         };
 
         if (required) {
-          setRequiredFields(prevValue =>
-            !prevValue.includes(name) ? [...prevValue, name] : prevValue,
-          );
+          setRequiredFields(() => Object.keys(validations.current));
```

Comment:
This doesn't seem quite right, there might be fields that are marked required that do not have a validation.

Instead, I think we want to both add and remove here:
```
if (required) setRequiredFields(prevValue => !prevValue.includes(name) ? [...prevValue, name] : prevValue);
else setRequiredFields(prevValue => prevValue.filter(v => v !== name));
```

Reply:
This does not work in the case when a field is unmounted.

Eg.,
  const defaultValue = {
    ampm: 'Field One',
    name1: '',
    name2: '',
  };
  const [value, setValue] = useState(defaultValue);
  const [selectedOption, setSelectedOption] = useState('Field One');

  const optionOne = (
    <FormField
      label="Name 1"
      required
      name="name1"
    >
      <TextInput name="name1" />
    </FormField>
  );

  const optionTwo = (
    <FormField
      label="Name 2"
      validate={[
        (val) => {
          if (val && val.length === 1)
            return { message: 'Text must be >1 character', status: 'error' };
          return undefined;
        },
        (val) => {
          if (val) return { message: 'Informational Status', status: 'info' };
          return undefined;
        },
      ]}
      tooltip="This information is publicly available !"
      required
      name="name2"
    >
      <TextInput name="name2" />
    </FormField>
  );

  return (
    <Grommet>
      <Form
        validate="change"
        onChange={(obj) => {
          setValue(obj);
          setSelectedOption(obj.ampm);
        }}
        value={value}
        onValidate={(validationResults) => {
          console.log('validationResults = ', validationResults);
        }}
      >
          <FormField
            label="Select a Form"
            name="ampm"
            required
          >
            <RadioButtonGroup name="ampm" options={['Field One', 'Field Two']} />
          </FormField>
          {selectedOption === 'Field One' ? optionOne : optionTwo}
      </Form>
    </Grommet>
  );

Reply:
I didn't quite follow the code snippets above. I think I'm getting it though, we need to `setRequiredFields()` in the case where fields are unmounted. I'm wondering if we should do this inside the effect at line 84, but outside of the call to `setValidationResults()` around line 96.

Reply:
I think `validations.current` would be better if placed in useEffect rather than in useFormField.
Reverting back to initial commit where required fields are filtered based on current validations.

Reply:
The filter can even be in validateField function.

        const validateField = (value2, data) => {
          let result;
          if (
            required &&
            // false is for CheckBox
            (value2 === undefined || value2 === '' || value2 === false)
          ) {
            result = messages.required;
          } else if (validateArg) {
            if (Array.isArray(validateArg)) {
              validateArg.some(aValidate => {
                result = validateSingle(aValidate, value2, data);
                return !!result;
              });
            } else {
              result = validateSingle(validateArg, value2, data);
            }
          }
          setRequiredFields(
            requiredFields.filter(n =>
              [Object.keys(validations.current)].includes(n),
            ),
          );
          return result;
        };

Reply:
Didnt even notice that am setting it inside `setValidationResults`.
Made the change, thanks !